packages/wallets/src/api/client.ts

```diff
@@ -0,0 +1,302 @@
+import { validateAPIKey, environmentToCrossmintBaseURL, APIKeyUsageOrigin } from "@crossmint/common-sdk-base";
+import { type Client, createClient, createConfig } from "@hey-api/client-fetch";
+
+import type {
+    ClientOptions,
+    CreateWalletDto,
+    WalletV1Alpha2ResponseDto,
+    CreateTransactionDto,
+    SubmitApprovalDto,
+    WalletsV1ControllerGetTransaction4Response,
+    WalletsV1Alpha2TransactionResponseDto,
+    CreateSignatureRequestDto,
+    WalletsV1Alpha2SignatureResponseDto,
+    WalletsV1Alpha2TransactionsResponseDto,
+    Nftevm,
+    Nftsol,
+    WalletBalanceResponseDto,
+} from "./gen/types.gen";
+import {
+    walletsV1ControllerCreateWallet4,
+    walletsV1ControllerGetWalletByLocator4,
+    walletsV1ControllerCreateTransaction4,
+    walletsV1ControllerSubmitApprovals4,
+    walletsV1ControllerGetTransaction4,
+    walletsV1ControllerCreateSignatureRequest4,
+    walletsV1ControllerSubmitSignatureApprovals4,
+    walletsV1ControllerGetSignature4,
+    walletsV1ControllerGetTransactionsWithoutChain4,
+    fetchContentFromWallet,
+    balanceControllerGetBalanceForLocator2,
+} from "./gen/sdk.gen";
+
+type CreateWalletParams = CreateWalletDto;
+type CreateWalletResponse = WalletV1Alpha2ResponseDto;
+type GetWalletResponse = WalletV1Alpha2ResponseDto;
+
+type CreateTransactionParams = CreateTransactionDto;
+type CreateTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+type ApproveTransactionParams = SubmitApprovalDto;
+type ApproveTransactionResponse = WalletsV1ControllerGetTransaction4Response;
+type GetTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+
+type CreateSignatureParams = CreateSignatureRequestDto;
+type CreateSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type ApproveSignatureParams = SubmitApprovalDto;
+type ApproveSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type GetSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+
+type GetTransactionsResponse = WalletsV1Alpha2TransactionsResponseDto;
+type GetNftsResponse = Nftevm | Nftsol;
+type GetBalanceResponse = WalletBalanceResponseDto;
+
+class ApiClient {
```

Comment:
I thought we had already something like a api client class in base SDK. Did you check? Would it be possible to reuse parts of that?

Reply:
ended up using `CrossmintApiClient` to make API calls in https://github.com/Crossmint/crossmint-sdk/pull/956/commits/9a34dd9f082adc0962ed5d5cc483adfaf8007f85

Reply:
Then we don't use the autogenerated low level calls and types..?

Reply:
Still using the types, just not the runtime code.